### PR TITLE
Upgrade UgandaEMR to OpenMRS 2.5.x

### PR DESCRIPTION
**Pull request by slubwama**
Upgrade UgandaEMR to OpenMRS 2.5.x
https://app.asana.com/0/1133167201254913/1202098258262134

### CODE REVIEW

**Comment by slubwama** [comment]
[dkayiwa](https://github.com/dkayiwa) I need your help on this one. Currently, all tests are failing and not sure what the exact cause is there is this error here https://pastebin.com/PjhT7LE0

**Comment by dkayiwa** [comment]
@slubwama change `BaseContextSensitiveTest` to `BaseModuleContextSensitiveTest`

**Comment by slubwama** [comment]
@dkayiwa same even after making the changes

**Comment by dkayiwa** [comment]
@slubwama did you finally figure this out?

**Comment by slubwama** [comment]
> @slubwama did you finally figure this out?

@dkayiwa still have issues with all tests failing due to null service calls and example stack trace is here https://pastebin.com/nFQDKX80

**Comment by dkayiwa** [comment]
@slubwama apply this git patch and compile again: https://pastebin.com/qzfcKqad

File: api/src/test/java/org/openmrs/module/ugandaemr/RolePrivilegeImportTest.java [hunk 1]
```diff
@@ -15,9 +16,16 @@ public class RolePrivilegeImportTest extends BaseModuleContextSensitiveTest {
 
     protected static final String ROLE_PRIVILLEGE_DATASET_XML = "metadata/Role_Privilege.xml";
 
+    UserService userService=null;
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/module/ugandaemr/RolePrivilegeImportTest.java]: @slubwama you do not need the above line if you do the changes in my patch.

Comment (slubwama) [reply api/src/test/java/org/openmrs/module/ugandaemr/RolePrivilegeImportTest.java]: @dkayiwa true and it has made the tests run well. though I am now getting a run time error  found in this stack trace https://pastebin.com/5xRPRyJr which causes the server to fail to start

Comment (dkayiwa) [reply api/src/test/java/org/openmrs/module/ugandaemr/RolePrivilegeImportTest.java]: This is the cause of the error `ClassNotFoundException: org.springframework.web.servlet.mvc.annotation.DefaultAnnotationHandlerMapping`. Just remove it from the spring configuration files of your modules.

Comment (slubwama) [reply api/src/test/java/org/openmrs/module/ugandaemr/RolePrivilegeImportTest.java]: @dkayiwa thanks. It has started Now going to test functionality in all modules

**Comment by solemabrothers** [comment]
@slubwama the UgandaEMR Module can not start with the 2.5.4-SNAPHOT version of openmrs. 

It throws the error. 

```
: Module requires version matching 2.5.4-SNAPSHOT.  Current code version is 2.3.1
	at org.openmrs.module.ModuleUtil.checkRequiredVersion(ModuleUtil.java:399)
	at org.openmrs.module.ModuleFactory.startModuleInternal(ModuleFactory.java:653)
	at org.openmrs.api.context.Daemon$1.run(Daemon.java:80)

```

**Comment by slubwama** [comment]
@

> @slubwama the UgandaEMR Module can not start with the 2.5.4-SNAPHOT version of openmrs.
> 
> It throws the error.
> 
> ```
> : Module requires version matching 2.5.4-SNAPSHOT.  Current code version is 2.3.1
> 	at org.openmrs.module.ModuleUtil.checkRequiredVersion(ModuleUtil.java:399)
> 	at org.openmrs.module.ModuleFactory.startModuleInternal(ModuleFactory.java:653)
> 	at org.openmrs.api.context.Daemon$1.run(Daemon.java:80)
> ```

@solemabrothers did you deploy the entire distro

File: pom.xml [hunk 2]
```diff
@@ -59,63 +59,65 @@
...
-		<registrationappVersion>1.18.0</registrationappVersion>
-		<registrationcoreVersion>1.9.0</registrationcoreVersion>
-		<reportingRestVersion>1.11.0</reportingRestVersion>
+		<legacyuiVersion>1.8.4</legacyuiVersion>
+		<metadatadeployVersion>1.13.0</metadatadeployVersion>
+		<metadatamappingVersion>1.4.0</metadatamappingVersion>
+		<metadatasharingVersion>1.8.0</metadatasharingVersion>
+		<openconceptlabVersion>1.2.9</openconceptlabVersion>
+		<openwebappsVersion>1.13.0</openwebappsVersion>
+		<patientFlagsVersion>3.0.3-SNAPSHOT</patientFlagsVersion>
```
Comment (solemabrothers) [inline pom.xml:100]: @slubwama lets depend on the latest released version of Patient Flags and Data Integrity modules

File: pom.xml [hunk 3]
```diff
@@ -59,63 +59,65 @@
...
+		<coreappsVersion>1.34.0</coreappsVersion>
 		<databaseBackupVersion>1.3.0</databaseBackupVersion>
 		<dataEntryStatisticsVersion>1.7.0</dataEntryStatisticsVersion>
-		<dataexchangeVersion>1.3.3</dataexchangeVersion>
-		<dataintegrityVersion>4.4.3</dataintegrityVersion>
-		<emrapiVersion>1.29.0</emrapiVersion>
-		<eventVersion>2.9.0</eventVersion>
-		<fhir2Version>1.2.2</fhir2Version>
+		<dataexchangeVersion>1.3.7</dataexchangeVersion>
+		<dataintegrityVersion>4.4.4-SNAPSHOT</dataintegrityVersion>
```
Comment (solemabrothers) [inline pom.xml:82]: @slubwama for your attention

File: pom.xml [hunk 4]
```diff
@@ -59,63 +59,65 @@
 	</modules>
 
 	<properties>
-		<openMRSVersion>2.3.1</openMRSVersion>
+		<openMRSVersion>2.5.4-SNAPSHOT</openMRSVersion>
```
Comment (solemabrothers) [inline pom.xml]: @slubwama update this to the latest released version

**Comment by solemabrothers** [comment]
@slubwama please change all snapshots to released versions

File: pom.xml (hunk 2 above)
Comment (solemabrothers) [reply pom.xml:100]: @slubwama please update these snapshots to the latest versions that have been released.

**Pull request closed without merging**